timeseries/src/autogluon/timeseries/models/ensemble/greedy_ensemble.py

```diff
@@ -38,6 +38,10 @@ def __init__(
 
     def _fit(self, predictions, labels, time_limit=None, sample_weight=None):
         self.dummy_pred = copy.deepcopy(predictions[0])
+        # This should never happen; sanity check to make sure that predict is implemented correctly for all models
```

Comment:
if it's a sanity check (the user cannot provide anything to break this), then we can replace with an `assert`. If the user can break this, we can replace with a `ValueError`. In any case I think we're fine without the TODO?

Reply:
Regarding the TODO: This assumption can break if we introduce new models that don't respect the item order or break the old ones.

The options are:
1. Keep an assert to catch this behavior at runtime & force us to fix the item order
2. Sort the items here to handle even incorrect item order

Probably 1 is the better option, but I wasn't sure

Reply:
Should be fixed now